hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientScheduledFutureProxy.java

```diff
@@ -81,24 +81,24 @@ public ScheduledTaskStatistics getStats() {
         String schedulerName = handler.getSchedulerName();
         String taskName = handler.getTaskName();
         int partitionId = handler.getPartitionId();
-        if (uuid != null) {
-            ClientMessage request = ScheduledExecutorGetStatsFromMemberCodec.encodeRequest(schedulerName, taskName, uuid);
-            ClientMessage response = invokeOnMember(request, uuid);
+
+        if (partitionId != -1) {
```

Comment:
I think you can leave the code as it is(as before your changes), and just change the check to `partitionId == -1`. also it might make sense to add asserts there to make sure that `uuid` is not null here.

Reply:
>add asserts there to make sure that uuid is not null here

I don't follow it.

```java
if (partitionId == -1) {
  // ...
} else if (uuid != null) {
  // ...
}
```

Something like that? Should there be a fallback `else` block? What would it do?

Reply:
Something like 
```java
        if (partitionId == -1) {
            assert uuid != null;
            
            ClientMessage request = ScheduledExecutorGetDelayFromMemberCodec.encodeRequest(schedulerName, taskName, uuid);
            ClientMessage response = invokeOnMember(request, uuid);
            long nanos = ScheduledExecutorGetDelayFromMemberCodec.decodeResponse(response);
            return unit.convert(nanos, TimeUnit.NANOSECONDS);
        } else {
            ClientMessage request = ScheduledExecutorGetDelayFromPartitionCodec.encodeRequest(schedulerName, taskName);
            ClientMessage response = invokeOnPartition(request, partitionId);
            long nanos = ScheduledExecutorGetDelayFromPartitionCodec.decodeResponse(response);
            return unit.convert(nanos, TimeUnit.NANOSECONDS);
        }
```

Not super nice, but almost the same one with the one we had before